src/v/reflection/async_adl.h

```diff
@@ -60,28 +152,4 @@ struct async_adl<std::optional<T>> {
     }
 };
 
-template<typename T>
-struct async_adl<std::vector<T>> {
-    using value_type = std::remove_reference_t<std::decay_t<T>>;
```

Comment:
are there any cases where these transforms on the value type of the container are important in the new refactored implementation?

Reply:
Can you clarify the question?

Reply:
`std::vector<T>` always had the value_type alias (which your PR is using). is there a reason that wasn't used before and instead the value type was derived via `std::remove_reference_t<std::decay_t<T>>;`

Reply:
Because now we get `value_type` and `key_type` from `typename List::value_type` and `typename Map::keyed_type` etc.

Reply:
Yes, we now get `value_type` etc... from the container's type aliases. Those type were available before your change, but the previous version didn't use them. Is there a use cases we're missing?

Reply:
looks good